iac-extensions/docker/src/test/java/org/sonar/iac/docker/tree/impl/AddTreeImplTest.java

```diff
@@ -0,0 +1,163 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.docker.tree.impl;
+
+import org.junit.jupiter.api.Test;
+import org.sonar.iac.docker.parser.grammar.DockerLexicalGrammar;
+import org.sonar.iac.docker.parser.utils.Assertions;
+import org.sonar.iac.docker.tree.api.AddTree;
+import org.sonar.iac.docker.tree.api.DockerTree;
+import org.sonar.iac.docker.tree.api.KeyValuePairTree;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.sonar.iac.common.testing.TextRangeAssert.assertTextRange;
+import static org.sonar.iac.docker.tree.impl.DockerTestUtils.parse;
+
+class AddTreeImplTest {
+  @Test
+  void matchingSimple() {
+    Assertions.assertThat(DockerLexicalGrammar.ADD)
+      .matches("ADD src dest")
+      .matches("add src dest")
+      .matches("ADD dest")
+      .matches("ADD src1 src2 src3 dest")
+      .matches("ADD --option src dest")
+      .matches("ADD --option=value src dest")
+      .matches("ADD --chown=55:mygroup files* /somedir/")
+      .matches("ADD --link /foo /bar")
+      .matches("ADD \"src\" \"dest\"")
+      .notMatches("ADD --option= src dest")
+      .notMatches("ADD--option= src dest")
+      .notMatches("ADDD --option= src dest")
+      .notMatches("ADD")
+      .notMatches("ADD ")
+      .notMatches("ADD --option=value");
+  }
+
+  @Test
+  void addInstructionSimple() {
+    AddTree tree = parse("ADD src dest", DockerLexicalGrammar.ADD);
+    assertThat(tree.getKind()).isEqualTo(DockerTree.Kind.ADD);
+    assertThat(tree.keyword().value()).isEqualTo("ADD");
+    assertTextRange(tree.textRange()).hasRange(1, 0, 1, 12);
+    assertThat(tree.options()).isEmpty();
+    assertThat(tree.srcs()).hasSize(1);
+    assertThat(tree.srcs().get(0).value()).isEqualTo("src");
+    assertThat(tree.dest().value()).isEqualTo("dest");
+  }
+
+  @Test
+  void addInstructionSimpleQuotes() {
+    AddTree tree = parse("ADD \"src\" \"dest\"", DockerLexicalGrammar.ADD);
+    assertThat(tree.getKind()).isEqualTo(DockerTree.Kind.ADD);
+    assertThat(tree.keyword().value()).isEqualTo("ADD");
+    assertTextRange(tree.textRange()).hasRange(1, 0, 1, 16);
+    assertThat(tree.options()).isEmpty();
+    assertThat(tree.srcs()).hasSize(1);
+    assertThat(tree.srcs().get(0).value()).isEqualTo("\"src\"");
+    assertThat(tree.dest().value()).isEqualTo("\"dest\"");
+  }
+
+  @Test
+  void addInstructionNoSrc() {
+    AddTree tree = parse("ADD dest", DockerLexicalGrammar.ADD);
+    assertThat(tree.getKind()).isEqualTo(DockerTree.Kind.ADD);
+    assertThat(tree.keyword().value()).isEqualTo("ADD");
```

Comment:
For the next time. For me, it would be okay, to only test what has changed, compared to the previous test. If we don't expect, that the kind and keyword changed, it's okay to skip this assertion.